src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java

```diff
@@ -200,28 +200,47 @@
                                                             List<CaseWorkerProfile> updateCaseWorkerProfiles,
                                                             Map<String, CaseWorkersProfileCreationRequest> requestMap) {
         List<CaseWorkerProfile> processedCwProfiles = null;
-        if (isNotEmpty(updateCaseWorkerProfiles)) {
-            caseWorkerLocationRepository.deleteByCaseWorkerProfileIn(updateCaseWorkerProfiles);
-            caseWorkerWorkAreaRepository.deleteByCaseWorkerProfileIn(updateCaseWorkerProfiles);
-            caseWorkerRoleRepository.deleteByCaseWorkerProfileIn(updateCaseWorkerProfiles);
+        // update roles in sidam and filter if failed in User profile
+        List<CaseWorkerProfile> filteredUpdateCwProfiles = updateSidamRoles(updateCaseWorkerProfiles, requestMap);
+
+        if (isNotEmpty(filteredUpdateCwProfiles)) {
+            caseWorkerLocationRepository.deleteByCaseWorkerProfileIn(filteredUpdateCwProfiles);
+            caseWorkerWorkAreaRepository.deleteByCaseWorkerProfileIn(filteredUpdateCwProfiles);
```

Comment:
Refactor all child deletes in single method and add some comments why we are doing that will be useful in future to maintain and also please mention right point of delete is after sidam updates.

Reply:
yes , anyways this code will be refactored in review comments change.